music21/expressions.py

```diff
@@ -870,6 +870,7 @@ def realize(
                                            inPlace=True)
 
             if self._setAccidentalFromKeySig:
+                assert currentKeySig is not None
```

Comment:
Incredibly unlikely that `self._setAccidentalFromKeySig` was toggled from False to True as a side effect of the last few lines since it was just checked a minute ago to get a value . But mypy doesn't know that.

Reply:
Better for this to assign a local variable `setAccidentalFromKeySig = self._setAccidentalFromKeySig` and then check against the local version; avoids another assert.